arelle/plugin/validate/ROS/rules/__init__.py

```diff
@@ -17,16 +18,41 @@ def errorOnRequiredFact(
         message: str,
 ) -> Iterable[Validation]:
     """
-    Yields an error if a fact with the given QName is not tagged with a non-nil value.
+    Yields an error if a fact with the given localName is not tagged with a non-nil value.
     :return: Yields validation errors.
     """
     facts: set[ModelFact] = modelXbrl.factsByLocalName.get(conceptLn, set())
     for fact in facts:
         if not fact.isNil:
             return
     yield Validation.error(
+        conceptLn=conceptLn,
         codes=code,
         msg=message,
-        conceptLn=conceptLn,
+
 
     )
+
+
+def errorOnPositiveOnlyFact(
+        modelXbrl: ModelXbrl,
+        conceptLn: str,
+        code: str,
+        message: str,
+) -> Iterable[Validation]:
+    """
+    Yields an error if a fact with the given localName is tagged with a negative value.
+    :return: Yields validation errors.
+    """
+    errorModelFacts: list[ModelFact | None] = []
```

Comment:
I dont believe that you need a None here for typing. The list will either be a list of of model facts or an empty list of model facts.